### YamlMime:ManagedReference
items:
- uid: System.SystemException
  id: SystemException
  children:
  - System.SystemException.#ctor
  - System.SystemException.#ctor(System.String)
  - System.SystemException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.SystemException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: SystemException
  nameWithType: SystemException
  fullName: System.SystemException
  type: Class
  summary: "用作系统异常命名空间的基类。"
  remarks: "此类用作一种区分系统异常和应用程序异常的方法。 它是为此类异常的基类<xref:System.ArgumentException>， <xref:System.FormatException>，和<xref:System.InvalidOperationException>。</xref:System.InvalidOperationException> </xref:System.FormatException> </xref:System.ArgumentException>      1> [!IMPORTANT]&1;> 由于 SystemException 用作不同的异常类型的基类，你的代码不应引发一个 SystemException 异常，也不应尝试处理 SystemException 异常，除非您想要重新引发原始异常。       SystemException 使用 HRESULT COR_E_SYSTEM，具有 0x80131501 的值。"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class SystemException : Exception
  inheritance:
  - System.Exception
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.SystemException.#ctor
  id: '#ctor'
  parent: System.SystemException
  langs:
  - csharp
  name: SystemException()
  nameWithType: SystemException.SystemException()
  fullName: System.SystemException.SystemException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "初始化的新实例<xref href=&quot;System.SystemException&quot;></xref>类。"
  remarks: "此构造函数初始化<xref:System.Exception.Message%2A>为系统提供的消息描述该错误，如&quot;系统错误。&quot;的新实例的属性</xref:System.Exception.Message%2A> 此消息将当前系统区域性考虑在内。       下表显示<xref:System.SystemException>.</xref:System.SystemException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|空引用 (`Nothing`在 Visual Basic 中)。 |  |<xref:System.Exception.Message%2A>|本地化的错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public SystemException ();
    parameters: []
  overload: System.SystemException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.SystemException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.SystemException
  langs:
  - csharp
  name: SystemException(String)
  nameWithType: SystemException.SystemException(String)
  fullName: System.SystemException.SystemException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "初始化的新实例<xref href=&quot;System.SystemException&quot;></xref>与指定的错误消息的类。"
  remarks: "内容`message`参数应为用户所理解。 此构造函数的调用方需要确保此字符串已针对当前系统区域性进行了本地化。       下表显示<xref:System.SystemException>.</xref:System.SystemException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|空引用 (`Nothing`在 Visual Basic 中)。 |  |<xref:System.Exception.Message%2A>|错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public SystemException (string message);
    parameters:
    - id: message
      type: System.String
      description: "描述错误的消息。"
  overload: System.SystemException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.SystemException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.SystemException
  langs:
  - csharp
  name: SystemException(SerializationInfo,StreamingContext)
  nameWithType: SystemException.SystemException(SerializationInfo,StreamingContext)
  fullName: System.SystemException.SystemException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "初始化的新实例<xref href=&quot;System.SystemException&quot;></xref>用序列化数据的类。"
  remarks: "在反序列化，以便重建通过流传输的异常对象过程中调用此构造函数。 有关详细信息，请参阅[XML 和 SOAP 序列化](~/add/includes/ajax-current-ext-md.md)。"
  syntax:
    content: protected SystemException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "承载序列化的对象数据的对象。"
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "有关源或目标的上下文信息。"
  overload: System.SystemException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.SystemException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.SystemException
  langs:
  - csharp
  name: SystemException(String,Exception)
  nameWithType: SystemException.SystemException(String,Exception)
  fullName: System.SystemException.SystemException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "初始化的新实例<xref href=&quot;System.SystemException&quot;></xref>使用指定的错误消息和对导致此异常的内部异常的引用的类。"
  remarks: "前一个异常的直接结果可以包含的引用中的前一个异常引发的异常，<xref:System.Exception.InnerException%2A>属性。</xref:System.Exception.InnerException%2A> <xref:System.Exception.InnerException%2A>属性返回相同的值传递到构造函数中，则为空引用 (`Nothing`在 Visual Basic 中) 如果<xref:System.Exception.InnerException%2A>属性不会提供给构造函数的内部异常值。</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       下表显示<xref:System.SystemException>.</xref:System.SystemException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|内部异常引用。 |  |<xref:System.Exception.Message%2A>|错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public SystemException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "解释异常原因的错误消息。"
    - id: innerException
      type: System.Exception
      description: "导致当前异常的异常。 如果`innerException`参数不是 null 引用 (<xref uid=&quot;langword_csharp_Nothing&quot; name=&quot;Nothing&quot; href=&quot;&quot;> </xref>在 Visual Basic 中) 中, 引发当前异常<xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;></xref>处理内部异常的块。"
  overload: System.SystemException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.Exception
  isExternal: false
  name: System.Exception
- uid: System.SystemException.#ctor
  parent: System.SystemException
  isExternal: false
  name: SystemException()
  nameWithType: SystemException.SystemException()
  fullName: System.SystemException.SystemException()
- uid: System.SystemException.#ctor(System.String)
  parent: System.SystemException
  isExternal: false
  name: SystemException(String)
  nameWithType: SystemException.SystemException(String)
  fullName: System.SystemException.SystemException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.SystemException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.SystemException
  isExternal: false
  name: SystemException(SerializationInfo,StreamingContext)
  nameWithType: SystemException.SystemException(SerializationInfo,StreamingContext)
  fullName: System.SystemException.SystemException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.SystemException.#ctor(System.String,System.Exception)
  parent: System.SystemException
  isExternal: false
  name: SystemException(String,Exception)
  nameWithType: SystemException.SystemException(String,Exception)
  fullName: System.SystemException.SystemException(String,Exception)
- uid: System.SystemException.#ctor*
  parent: System.SystemException
  isExternal: false
  name: SystemException
  nameWithType: SystemException.SystemException
